src/Disks/DiskObjectStorage/MetadataStorages/PlainRewritable/MetadataStorageFromPlainRewritableObjectStorage.cpp

```diff
@@ -424,13 +407,16 @@ void MetadataStorageFromPlainRewritableObjectStorageTransaction::createDirectory
         return;
     }
 
+    if (!uncommitted_fs_tree->getDirectoryRemoteInfo(path))
+        uncommitted_fs_tree->recordDirectoryPath(path, DirectoryRemoteInfo{ .remote_path = getRandomASCIIString(32), .etag = "", .files = {}});
+
```

Comment:
Currently the change affects creation and deletion, will it be extended to, e.g., `moveDirectory`?